src/mlpack/tests/main_tests/dbscan_test.cpp

```
@@ -0,0 +1,517 @@
+ */
+BOOST_AUTO_TEST_CASE(DBSCANOutputDimensionTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  size_t inputSize = inputData.n_cols;
+
+  SetInputParam("input", inputData);
+
+  mlpackMain();
+
+  // Check that number of predicted labels is equal to the input test points.
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Row<size_t>>("assignments").n_cols,
+                      inputSize);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Row<size_t>>("assignments").n_rows,
+                      1);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroids").n_rows, 4);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroids").n_cols, 2);
+}
+
+/**
+ * Check that radius of search(epsilon) is always non-negative.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANEpsilonTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) -0.5);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Check that minimum size of cluster is always non-negative.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANMinSizeTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("min_size", (int) -1);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Number of clusters should be less than the input number of points
+ * when min_size is equal to 0.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANClusterNumberTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("min_size", (int) 0);
```

I'm a little confused by this---the check you added to `dbscan_main.cpp` forces `min_size` to be greater than 0.  So `mlpackMain()` should throw an exception here.  I'm also not sure I understand what you mean to test here; can you clarify please?

---

I am sorry for that. It should be >= 0 in `dbscan_main.cpp`
Setting min_size = 0 ensures that there will be no noise points which will maximize the number of clusters.

---

If I am not mistaken, `min_size = 1` should also guarantee there are no noise points.

---

Yes, you're right. I'm sorry about the confusion!